cmd/node/app_test.go

```diff
@@ -2,70 +2,128 @@ package node
 
 import (
 	"fmt"
+	"github.com/spacemeshos/go-spacemesh/activation"
 	"github.com/spacemeshos/go-spacemesh/address"
 	apiCfg "github.com/spacemeshos/go-spacemesh/api/config"
+	"github.com/spacemeshos/go-spacemesh/crypto"
+	"github.com/spacemeshos/go-spacemesh/database"
 	"github.com/spacemeshos/go-spacemesh/eligibility"
 	"github.com/spacemeshos/go-spacemesh/log"
+	"github.com/spacemeshos/go-spacemesh/mesh"
 	"github.com/spacemeshos/go-spacemesh/miner"
+	"github.com/spacemeshos/go-spacemesh/nipst"
 	"github.com/spacemeshos/go-spacemesh/oracle"
 	"github.com/spacemeshos/go-spacemesh/p2p/service"
 	"github.com/spacemeshos/go-spacemesh/signing"
-	sync2 "github.com/spacemeshos/go-spacemesh/sync"
 	"github.com/spacemeshos/go-spacemesh/types"
+	"github.com/spacemeshos/poet/integration"
+	"github.com/stretchr/testify/require"
 	"github.com/stretchr/testify/suite"
 	"math/big"
 	"os"
+	"path/filepath"
 	"sync"
 	"testing"
 	"time"
-
-	"github.com/stretchr/testify/assert"
 )
 
 type AppTestSuite struct {
 	suite.Suite
 
-	apps []*SpacemeshApp
-	dbs  []string
+	apps        []*SpacemeshApp
+	dbs         []string
+	poetCleanup func() error
 }
 
 func (app *AppTestSuite) SetupTest() {
 	app.apps = make([]*SpacemeshApp, 0, 0)
 	app.dbs = make([]string, 0, 0)
 }
 
-func (app *AppTestSuite) TearDownTest() {
+// NewRPCPoetHarnessClient returns a new instance of RPCPoetClient
+// which utilizes a local self-contained poet server instance
+// in order to exercise functionality.
+func NewRPCPoetHarnessClient() (*nipst.RPCPoetClient, error) {
+	h, err := integration.NewHarness()
+	if err != nil {
+		return nil, err
+	}
+
+	return nipst.NewRPCPoetClient(h.PoetClient, h.TearDown), nil
+}
 
+func (app *AppTestSuite) TearDownTest() {
+	if err := app.poetCleanup(); err != nil {
+		log.Error("error while cleaning up PoET: %v", err)
+	}
 	for _, dbinst := range app.dbs {
-		err := os.RemoveAll(dbinst)
-		if err != nil {
+		if err := os.RemoveAll(dbinst); err != nil {
 			panic(fmt.Sprintf("what happened : %v", err))
 		}
 	}
+	if err := os.RemoveAll("../tmp"); err != nil {
+		log.Error("error while cleaning up tmp dir: %v", err)
+	}
+	if matches, err := filepath.Glob("*.bin"); err != nil {
+		log.Error("error while finding PoET bin files: %v", err)
```

Comment:
The poet should clean up after himself. See https://github.com/spacemeshos/poet/issues/35.

If you want to keep this in the meanwhile, add a comment.